examples/file-sharing/Cargo.toml

```diff
@@ -6,12 +6,20 @@ publish = false
 license = "MIT"
 
 [dependencies]
+libp2p-request-response = { workspace = true, features = ["cbor"]  }
+libp2p-swarm = { workspace = true }
+serde = { version = "1.0", optional = true, features = ["derive"] }
+serde_cbor = { version = "0.11.2", optional = true }
 async-std = { version = "1.12", features = ["attributes"] }
 async-trait = "0.1"
 clap = { version = "4.3.1", features = ["derive"] }
 either = "1.8"
 env_logger = "0.10"
 futures = "0.3.28"
-libp2p = { path = "../../libp2p", features = ["async-std", "dns", "kad", "noise", "macros", "request-response", "tcp", "websocket", "yamux"] }
+libp2p = { path = "../../libp2p", features = ["async-std", "dns", "kad", "noise", "macros", "request-response", "tcp", "websocket", "yamux"]}
 multiaddr = { version = "0.17.1" }
 void = "1.0.2"
+
+[features]
+default = ["cbor"]
+cbor = ["dep:serde", "dep:serde_cbor", "libp2p-swarm/macros"]
```

Comment:
Do we need to feature-flag this? Wouldn't the example be easier to understand without?

Reply:
@thomaseizinger Done.